source/Hardpoint.cpp

```diff
@@ -124,20 +124,20 @@ int Hardpoint::BurstRemaining() const
 
 
 
-// Perform one step (i.e. decrement the reload count).
-void Hardpoint::Step()
+// Perform many steps (i.e. decrement the reload count when the date changes).
+void Hardpoint::Step(int frames)
 {
 	if(!outfit)
 		return;
 	
 	wasFiring = isFiring;
 	if(reload > 0.)
-		--reload;
+		reload -= frames;
```

Comment:
please consider `reload -= min(reload, frames);` here.
Otherwise reload gets deeply negative, which is not corrected anywhere in Hardpoint.cpp
Same for the burstReload, please.

Hardpoint::Fire(...)
	// Reset the reload count.
	reload += outfit->Reload();
	burstReload += outfit->BurstReload();
	--burstCount;

--> you could fire a pretty heavy salvo after reloading a full day long in advance :-D